### PR TITLE
Add ecotax in product variation attributes (38070)

### PR DESCRIPTION
**Pull request by yves202**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | add ecotax in product variation attributes.
| Type?         | improvement
| BC breaks?    |no
| Deprecations? |no
| Fixed ticket? | Fixes #38070
| How to test?  |

### CODE REVIEW

**Comment by clotairer** [comment]
il faudrait nettoyer l’ancienne PR. 
Par exemple : ligne 111
[src/Services/ProductAttributeCombination.php](https://github.com/shoppingflux/module-prestashop/blob/23dc194d3fdf49de35bb7173ef9412037a82b074/src/Services/ProductAttributeCombination.php)

File: src/Services/ProductAttributeCombination.php [hunk 1]
```diff
@@ -108,7 +108,6 @@ public function get(Product $product, $idLang = null, $idShop = null)
                 $combinations[$idProductAttribute]['weight'] = $row['weight'];
                 $combinations[$idProductAttribute]['reference'] = $row['reference'];
                 $combinations[$idProductAttribute]['wholesale_price'] = $row['wholesale_price'];
-                $combinations[$idProductAttribute]['ecotax'] = $row['ecotax'];
```
Comment (clotairer) [inline src/Services/ProductAttributeCombination.php:111]: Si tu le supprimes ici, tu vas pas le retrouver dans ProductSerializer.php, non ?

**Review by clotairer — CHANGES_REQUESTED** [review]

File: src/Services/ProductAttributeCombination.php (hunk 1 above)
Comment (yves202) [reply src/Services/ProductAttributeCombination.php:111]: on le retroluvera jamais, la pull request est pas accepter,
https://github.com/shoppingflux/module-prestashop/blob/develop/controllers/front/product.php#L170

File: src/Services/ProductSerializer.php [hunk 2]
```diff
@@ -405,12 +405,16 @@ protected function getVariations($carrier, $productLink)
...
                 ],
-                'ecotax' => $combination['ecotax'],
             ];
 
             if (empty($combination['ean13']) === false) {
                 $variation['gtin'] = $combination['ean13'];
             }
+
+            if (empty($combination['ecotax']) === false) {
+                $variation['attributes']['ecotax_child'] = $combination['ecotax'];
```
Comment (clotairer) [inline src/Services/ProductSerializer.php:415]: possible de couvrir pa run test unitaire ?

**Review by clotairer — CHANGES_REQUESTED** [review]

**Review by clotairer — APPROVED** [review]

**Review by clotairer — APPROVED** [review]

**Pull request merged**